pkg/ccl/crosscluster/logical/logical_replication_job.go

```diff
@@ -308,14 +310,25 @@ func (p *logicalReplicationPlanner) generatePlanWithFrontier(
 	}
 
 	dstToSrcDescMap := make(map[int32]descpb.TableDescriptor)
-	tableIDs := make([]int32, len(p.payload.ReplicationPairs))
-	for i, pair := range p.payload.ReplicationPairs {
-		dstToSrcDescMap[pair.DstDescriptorID] = plan.DescriptorMap[pair.SrcDescriptorID]
-		tableIDs[i] = pair.DstDescriptorID
+	tableIDToPrefixMap := make(map[int32]string)
+
+	if err := sql.DescsTxn(ctx, execCfg, func(ctx context.Context, txn isql.Txn, descriptors *descs.Collection) error {
+		for _, pair := range p.payload.ReplicationPairs {
```

Comment:
nit: could you leave a todo here along the lines of: "combine tableIDToPrefixMap and dstToSrcDescMap"